test/runFIT.sh

```diff
@@ -0,0 +1,182 @@
+#!/bin/bash -e
+
+############################################
+# Global variable
+############################################
+RACKHD_TEST_DIR="$( cd -P "$( dirname "$0" )" && pwd )"
+
+#############################################
+#
+# Usage
+############################################
+Usage(){
+    set +x
+    echo "Function: This script is used to set up environment for FIT and run FIT."
+    echo "Usage: $0 [OPTIONS]"
+    echo "  OPTIONS:"
+    echo "    Mandatory options:"
+    echo "      -p, --PASSWORD: password of current user which is used for FIT to log into host by, it's required."
+    echo "    Optional options:"
+    echo "      -g, --TEST_GROUP: test group of FIT, such as imageservice, smoke"
+    echo "      -s, --TEST_STACK: target test stack of FIT, such as docker, vagrant..."
+    echo "      -v, --TEST_LOG_LEVEL: log level of FIT, 0 is the most verbose for log output , 9 for most succinct"
+    echo"       -w, --WORKSPACE: The directory of workspace( where the test report will be )"
+    echo "      -e, --TEST_EXTRA_OPTIONS: The extra options of FIT, such as '-extra myconfig.json -config my-dir -nodeid abcd'"
+    set -x
+}
+
+
+#############################################
+#
+#  Create the virtual env for FIT  
+#
+############################################
+setupVirtualEnv(){
+    pushd ${RACKHD_TEST_DIR}
+    virtual_env_name=FIT
+    rm -rf .venv/$virtual_env_name
+    ./mkenv.sh $virtual_env_name
+    source myenv_$virtual_env_name
+    popd
+}
+
+####################################
+#
+# 1. Modify FIT config files , to  using actual DHCP Host IP instead of 172.31.128.1
+#
+##################################
+setupTestsConfig(){
+    echo "SetupTestsConfig ...replace the 172.31.128.1 IP in test configs with actual DHCP port IP"
+    RACKHD_DHCP_HOST_IP=$(ifconfig | awk '/inet addr/{print substr($2,6)}' |grep 172.31.128)
+    if [ "$RACKHD_DHCP_HOST_IP" == "" ]; then
+         echo "[Error] There should be a NIC with 172.31.128.xxx IP in your OS."
+         exit -2
+    fi
+    pushd ${RACKHD_TEST_DIR}/config
+    sed -i "s/\"username\": \"vagrant\"/\"username\": \"${USER}\"/g" credentials_default.json
+    sed -i "s/\"password\": \"vagrant\"/\"password\": \"${PASSWORD}\"/g" credentials_default.json
+    popd
+    pushd ${RACKHD_TEST_DIR}
+    find ./ -type f -exec sed -i -e "s/172.31.128.1/${RACKHD_DHCP_HOST_IP}/g" {} \;
+    popd
+}
+
+####################################
+#
+# Collect the test report
+#
+##################################
+collectTestReport()
+{
+    if [ -n "$WORKSPACE" ]; then
+        mkdir -p ${WORKSPACE}/xunit-reports
+        cp ${RACKHD_TEST_DIR}/*.xml ${WORKSPACE}/xunit-reports
```

Comment:
If we moved *xml instead of copying *xml,  would it make the workspace a little cleaner?